presto-main/src/main/java/com/facebook/presto/operator/DistinctLimitOperator.java

```diff
@@ -178,6 +214,10 @@ public void addInput(Page page)
     public Page getOutput()
     {
         if (unfinishedWork != null && !processUnfinishedWork()) {
+            if (isTimedOut()) {
+                finish();
```

Comment:
isTimedOut method already calls finish, this seems like an extra call.

Reply:
Done